python/essex-config/README.md

```diff
@@ -0,0 +1,106 @@
+# essex-configuration
+
+essex-configuration is a Python library for creating configuration objects that read configuration from different sources. This library supports reading from files, environment variables, and Azure Key Vault.
+
+## Installation
+
+To install the essex-configuration library, use pip:
+
+`pip install essex-configuration`
+
+## Usage
+
+Here’s an example of how to use the essex-configuration library to create a configuration object for connecting to a customer database:
+
+```python
+from typing import Annotated, ClassVar
+from essex_config import Config, ConfigurationField, FieldType
+from essex_config.sources import Source, EnvSource, FileSource, KeyvaultSource
+
+
+class CustomerDatabase(Config):
+    """Class that holds the configuration to connect to the Customer Database."""
+
+    sources: ClassVar[list[Source]] = [
+        FileSource(file_path="pyproject.toml"),
+        EnvSource(),
+        FileSource(file_path="settings.json"),
+        KeyvaultSource("my-keyvault")
+    ]
+
+    host: Annotated[str, ConfigurationField(
+        alt_name="customer.host", description="Host ip or url to connect to the database",
```

Comment:
With sub-objects, will dot-notation be assumed to traverse sub-objects? If I define `database.password` in keyvault, will it land on the Database object's `password` field?

Reply:
Right now, the logic to figure out the possible name of a variable is the following:

For each defined source in the class that inherits `Config`:

1. Check if the name of the field is in the source and return the value.
2. Check if the alternative name is in the source (if the alt-name has `.` then try and dive deeper into the source dictionary)
3. For each fallback name check if it is in the source (if the alt-name has `.` then try and dive deeper into the source dictionary)

So, if your alt_name is "database.password" it would land in the password field, but if I understood correctly, you would like another step that checks for `cls.__name__ + "." + field_name`?